src/stores.ts

```diff
@@ -64,3 +71,25 @@ export function deleteIntervention(id: number) {
   sidebarHover.set(null);
   openFromSidebar.set(null);
 }
+
+function loadUserSettings(): UserSettings {
+  let settings = {
+    streetViewImagery: "google",
+    avoidDoublingBack: false,
+  };
+
+  // Be paranoid when loading from local storage, and only copy over valid items
+  try {
+    let x = JSON.parse(window.localStorage.getItem("userSettings") || "{}");
+    if (x.streetViewImagery == "google" || x.streetViewImagery == "bing") {
```

Comment:
Is there a way to this a bit more naturally? Create a `StreetViewProvider` type or similar? and then something like
`if (type x is == "StreetViewProvider")`

Reply:
Yeah looks good to me!